core/src/main/java/apoc/cypher/Cypher.java

```diff
@@ -242,6 +243,21 @@ public static String withParamMapping(String fragment, Collection<String> keys)
         return runCypherQuery(tx, statement, params);
     }
 
+    @Procedure(mode = WRITE)
+    @Description("apoc.cypher.runWrite(statement, params) yield value - alias for apoc.cypher.doIt")
+    public Stream<MapResult> runWrite(@Name("cypher") String statement, @Name("params") Map<String, Object> params) {
+        return doIt(statement, params);
+    }
+
+    @Procedure(mode = SCHEMA)
+    @Description("apoc.cypher.runSchema(statement, params) yield value - executes query schema statement with the given parameters")
+    public Stream<MapResult> runSchema(@Name("cypher") String statement, @Name("params") Map<String, Object> params) {
+        if (!isSchemaOperation(statement)) {
```

Comment:
I know this function was already there, but parsing Cypher is considered a bad practice, I'd remove the check.

What if there is no explicit `create/drop index/constraint` in the statement?
It could delegate to another procedure doing schema operations e.g.

Reply:
@fbiville It's strange, but removing this condition, the `testFail(db, "CALL apoc.cypher.runSchema('CREATE (n:Test {a: 1})',{})", QueryExecutionException.class);` in `CypherTest.java` doesn't work.
It seems like the `mode = SCHEMA` considers also WRITE operations.

To avoid duplicate code, maybe I can wait for the merge of the `validateQuery` that I created in this other pr
https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/1843/files#diff-0a04de30ec5509e502658b09de532ed90b645feae019d677dd83eb9e81a061db

In this way, with `validateQuery(db, statement, SCHEMA_WRITE)` it works. Wdyt?

Reply:
@JMHReif and I just double-checked Neo4j RBAC https://neo4j.com/docs/operations-manual/current/authentication-authorization/access-control/ and it seems that the default `architect` has schema **and** read/write privileges.

It seems technically possible to define a custom role with only schema privileges, but how common is that?
Does it make sense to enforce that `runSchema` can be run only with schema-only statements?

If you think so, the approach you took in the other PR makes sense (i.e. using `EXPLAIN`).

In short: should we drop the check or rely on `EXPLAIN`?

Reply:
Thinking about it.. it makes little sense to impose this restriction.
Ok, I removed the `isSchemaOperation`